src/Umbraco.Web.UI.Client/src/common/directives/components/umbmediagrid.directive.js

```diff
@@ -120,7 +120,7 @@ Use this directive to generate a thumbnail grid of media items.
                     setOriginalSize(item, itemMaxHeight);
 
                     // remove non images when onlyImages is set to true
-                    if(scope.onlyImages === "true" && !item.isFolder && !item.thumbnail){
+                    if (scope.onlyImages === "true" && item.isFolder === true) {
```

Comment:
I think this should be:
if (scope.onlyImages === "true" && item.isFolder === false && item.thumbnail == null) {